### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by ochirkov**

### CODE REVIEW

File: backuper/modules/cloud/amazon/elasticache.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+from time import sleep
+from backuper.modules.cloud.amazon import get_amazon_client
+from backuper.utils import get_msg
+from backuper.utils.constants import amazonRegions
+from backuper.utils.validate import ValidateBase
+
+
+class ValidateElasticache(ValidateBase):
+
+    def params_validate(self, **kwargs):
```
Comment (ochirkov) [inline backuper/modules/cloud/amazon/elasticache.py]: add engine param to parameters in action file and validate all possible parameters depends on different actions

File: backuper/modules/cloud/amazon/rds.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from backuper.modules.cloud.amazon import get_amazon_client
 from backuper.utils.validate import ValidateBase, validate_empty_snapshots
 from backuper.utils import get_msg
-from backuper.utils.constants import amazon_regions, wait_timeout
```
Comment (ochirkov) [inline backuper/modules/cloud/amazon/rds.py]: no camel case... it's not necessary

File: backuper/modules/cloud/amazon/rds.py [hunk 3]
```diff
@@ -12,251 +12,176 @@ class ValidateRDS(ValidateBase):
...
 
         if kwargs['action'] == 'create':
-
             parameters_schema = self.tr.Dict({
-                self.tr.Key('snapshot_identifier'): self.tr.String,
-                self.tr.Key('region'): self.tr.Enum(*amazon_regions),
-                self.tr.Key('db_identifier'): self.tr.String,
-                self.tr.Key('copy_to_region', optional=True): self.tr.List(
-                    self.tr.String, min_length=1),
-                self.tr.Key('wait_timeout', optional=True): self.tr.Int
```
Comment (ochirkov) [inline backuper/modules/cloud/amazon/rds.py]: where optional parameters from schema??????

File: backuper/modules/cloud/amazon/rds.py [hunk 4]
```diff
@@ -12,251 +12,176 @@ class ValidateRDS(ValidateBase):
...
-            self.validate.filters_validate(**self.kwargs)
-
-        return params
-
-    def get_snapshots(self, region):
-
-        c = get_amazon_client(self.config()['a_type'], region)
-        response = c.describe_db_snapshots()
+    def get_snapshots(self):
+        response = self.client.describe_db_snapshots(Engine=self.parameters['engine'])
```
Comment (ochirkov) [inline backuper/modules/cloud/amazon/rds.py]: use flake8 for python code lint

File: backuper/utils/validate.py [hunk 5]
```diff
@@ -49,7 +48,8 @@ def filters_validate(self, **kwargs):
             checks[filter['type']](filter)
 
 
-def validate_empty_snapshots(snapshots, msg):
```
Comment (ochirkov) [inline backuper/utils/validate.py]: return msg arg
via msg we can customize verbosity of exeption

File: backuper/utils/validate.py [hunk 6]
```diff
@@ -49,7 +48,8 @@ def filters_validate(self, **kwargs):
...
+def validate_empty_snapshots(snapshots):
 
     if not snapshots:
-        raise BackuperNoSnapshotMatchError(msg)
```
Comment (ochirkov) [inline backuper/utils/validate.py]: return exception

File: backuper/utils/validate.py [hunk 7]
```diff
@@ -49,7 +48,8 @@ def filters_validate(self, **kwargs):
...
+        # raise BackuperNoSnapshotMatchError(msg)
+        print("No snapshots were found.")
```
Comment (ochirkov) [inline backuper/utils/validate.py]: exception does print message depends on msg param
don't just print

File: backuper/modules/mongodb/mongodb.py [hunk 8]
```diff
@@ -39,7 +39,7 @@ def config(self):
                       a_collection=None,
                       a_gzip=None,
                       a_path=None,
-                      a_wait_timeout=None)
```
Comment (ochirkov) [inline backuper/modules/mongodb/mongodb.py]: no camel case!!!

File: backuper/modules/cloud/amazon/rds.py [hunk 9]
```diff
@@ -12,251 +12,176 @@ class ValidateRDS(ValidateBase):
...
-
             parameters_schema = self.tr.Dict({
-                self.tr.Key('snapshot_identifier'): self.tr.String,
-                self.tr.Key('region'): self.tr.Enum(*amazon_regions),
-                self.tr.Key('db_identifier'): self.tr.String,
-                self.tr.Key('copy_to_region', optional=True): self.tr.List(
-                    self.tr.String, min_length=1),
-                self.tr.Key('wait_timeout', optional=True): self.tr.Int
+                self.tr.Key('region'): self.tr.Enum(*amazonRegions),
+                self.tr.Key('engine'): self.tr.String,
```
Comment (ochirkov) [inline backuper/modules/cloud/amazon/rds.py]: engine should not be just a string
it is a list with allowed values (like postgresql, mysql and etc...)

File: backuper/modules/cloud/amazon/rds.py [hunk 10]
```diff
@@ -12,251 +12,176 @@ class ValidateRDS(ValidateBase):
...
         if kwargs['action'] == 'delete':
-
             parameters_schema = self.tr.Dict({
-                self.tr.Key('region'): self.tr.Enum(*amazon_regions),
-                self.tr.Key('snapshot_type'): self.tr.Enum(
+                self.tr.Key('region'): self.tr.Enum(*amazonRegions),
+                self.tr.Key('engine'): self.tr.String,
+                self.tr.Key('snapshotId'): self.tr.String,
+                self.tr.Key('snapshotType'): self.tr.Enum(
                     *['standard', 'manual', 'all'])
```
Comment (ochirkov) [inline backuper/modules/cloud/amazon/rds.py]: move list to constants.py

File: backuper/modules/cloud/amazon/rds.py (hunk 4 above)
Comment (ochirkov) [reply backuper/modules/cloud/amazon/rds.py]: add flake8 to requirements
add precommit hook to your local repo

File: backuper/modules/cloud/amazon/elasticache.py (hunk 1 above)
Comment (blckjack) [reply backuper/modules/cloud/amazon/elasticache.py]: no need in engine param in elasticache, due to aws restrictions.

File: backuper/modules/cloud/amazon/rds.py (hunk 2 above)
Comment (blckjack) [reply backuper/modules/cloud/amazon/rds.py]: done

File: backuper/modules/mongodb/mongodb.py (hunk 8 above)
Comment (blckjack) [reply backuper/modules/mongodb/mongodb.py]: done

File: backuper/modules/cloud/amazon/rds.py (hunk 9 above)
Comment (blckjack) [reply backuper/modules/cloud/amazon/rds.py]: done. created
engine: {rds:[], ...}

File: backuper/modules/cloud/amazon/rds.py (hunk 10 above)
Comment (blckjack) [reply backuper/modules/cloud/amazon/rds.py]: done

File: backuper/modules/cloud/amazon/rds.py (hunk 3 above)
Comment (blckjack) [reply backuper/modules/cloud/amazon/rds.py]: done

**Pull request merged**